lib/identifiers/isbn.rb

```diff
@@ -117,5 +115,34 @@ def self.valid_isbn_10?(isbn)
     def self.digits_of(isbn)
       isbn.to_s.each_char.map { |char| char == 'X' ? 10 : Integer(char) }.to_enum
     end
+
+    def self.isbn_with_prefix_candidates(str, prefix_regexp, isbn_regexp)
+      regexp = Regexp.new("#{prefix_regexp}#{isbn_regexp}", Regexp::IGNORECASE | Regexp::EXTENDED)
+
+      str
+        .to_s
+        .scan(regexp)
+        .filter_map do |match|
+          extract(Array(match).first)&.first
+        end
+    end
+
+    def self.generate_prefix_regexp(prefixes)
+      joined_prefixes = Regexp.union(prefixes).source
+
+      Regexp.new(
+        "(?<=                 # Lookbehind for a prefix
+           #{joined_prefixes} # ie:p1|p2|p3
+         )
+         :?                   # Optional colon. If you want to use a different separator, you can add it as a prefix
+         \\s*                 # Optional whitespaces
+         ", Regexp::IGNORECASE | Regexp::EXTENDED
+      )
+    end
+
+    def self.isbn_a_candidate_matcher
+      # We capture the ISBN-A prefix for the ISBN-A regexp to work correctly when extracting ISBN-As
+      Regexp.new(ISBN_A_REGEXP.source.gsub('(?<=10\\.)', '10\.').to_s, Regexp::IGNORECASE | Regexp::EXTENDED)
```

Comment:
`gsub` returns a string, why is `to_s` needed?

Reply:
To be honest, I don't remember adding it and it doesn't make sense to me neither 😅 

Removed here! 5d0e33de6e1e5cb1d6792683f9ac66f936bb9d2e